### PR TITLE
Match files using a regular expression pattern

### PR DESCRIPTION
**Pull request by thebinarymutant**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -14,5 +14,6 @@ jobs:
         id: migrations
         with:
           files: '["migrations/test.sql", "package.json"]'
+          pattern: 'migrations'
```
Comment (debugger22) [inline .github/workflows/ci.yml:17]: Shouldn't the pattern be `**migrations**`?

File: action.yml [hunk 2]
```diff
@@ -8,6 +8,9 @@ inputs:
   files:
     description: 'File paths'
     required: true
+  pattern:
+    description: 'File pattern to match using RegEx'
+    required: true
```
Comment (debugger22) [inline action.yml:13]: It shouldn't be a required param

File: .github/workflows/ci.yml (hunk 1 above)
Comment (thebinarymutant) [reply .github/workflows/ci.yml]: No, that is the pattern matching syntax used in shells

File: action.yml (hunk 2 above)
Comment (thebinarymutant) [reply action.yml]: Yes, corrected this one

**Pull request merged**